mountpoint-s3-fs/src/config.rs

```diff
@@ -0,0 +1,97 @@
+use anyhow::Context as _;
+use futures::executor::block_on;
+use mountpoint_s3_client::ObjectClient;
+
+use crate::data_cache::{DataCacheConfig, DiskDataCache, ExpressDataCache, MultilevelDataCache};
+use crate::fuse::config::FuseSessionConfig;
+use crate::fuse::session::FuseSession;
+use crate::fuse::S3FuseFilesystem;
+use crate::prefetch::{Prefetcher, PrefetcherBuilder};
+use crate::s3::config::S3Path;
+use crate::sync::Arc;
+use crate::{Runtime, S3Filesystem, S3FilesystemConfig};
+
+/// Configuration for a Mountpoint session
+#[derive(Debug)]
+pub struct MountpointConfig {
+    fuse_session_config: FuseSessionConfig,
+    data_cache_config: DataCacheConfig,
+    filesystem_config: S3FilesystemConfig,
+}
+
+impl MountpointConfig {
+    pub fn new(fuse_session_config: FuseSessionConfig) -> Self {
+        Self {
+            fuse_session_config,
+            data_cache_config: Default::default(),
+            filesystem_config: Default::default(),
+        }
+    }
+
+    pub fn filesystem_config(mut self, filesystem_config: S3FilesystemConfig) -> Self {
+        self.filesystem_config = filesystem_config;
+        self
+    }
+
+    pub fn data_cache_config(mut self, data_cache_config: DataCacheConfig) -> Self {
+        self.data_cache_config = data_cache_config;
+        self
+    }
+
+    /// Create a new FUSE session
+    pub fn create_fuse_session<Client>(
+        self,
+        s3_path: S3Path,
+        client: Client,
+        runtime: Runtime,
+    ) -> anyhow::Result<FuseSession>
+    where
+        Client: ObjectClient + Clone + Send + Sync + 'static,
+    {
+        let prefetcher_builder = create_prefetcher_builder(self.data_cache_config, &client, &runtime)?;
+        tracing::trace!(filesystem_config=?self.filesystem_config, "creating file system");
+        let fs = S3Filesystem::new(
+            client,
+            prefetcher_builder,
+            runtime,
+            &s3_path.bucket_name,
+            &s3_path.prefix,
+            self.filesystem_config,
+        );
+
+        let fuse_fs = S3FuseFilesystem::new(fs);
+        FuseSession::new(fuse_fs, self.fuse_session_config)
+    }
+}
+
+fn create_prefetcher_builder<Client>(
```

Comment:
Should this be a method of `create_prefetcher_builder`?

Reply:
Not sure what you mean. It's a private helper function for now. We may consider exposing it directly at some point and review where it should live then.